File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran-skills/tsvc_2_s2275__submitted.f90
subroutine tsvc_2_s2275_fp64(a, aa, b, bb, c, cc, d, LEN_2D) bind(C)
    use iso_c_binding
    implicit none
    integer(c_int64_t), value, intent(in) :: LEN_2D
    real(c_double), intent(inout) :: a(LEN_2D)
    real(c_double), intent(inout) :: aa(LEN_2D, LEN_2D)
    real(c_double), intent(in) :: b(LEN_2D)
    real(c_double), intent(in) :: bb(LEN_2D, LEN_2D)
    real(c_double), intent(in) :: c(LEN_2D)
    real(c_double), intent(in) :: cc(LEN_2D, LEN_2D)
    real(c_double), intent(in) :: d(LEN_2D)
    integer(c_int64_t) :: i, j

    !$omp parallel do private(i) schedule(static)
    do j = 1, LEN_2D
        !$omp simd
        do i = 1, LEN_2D
            aa(i,j) = aa(i,j) + bb(i,j) * cc(i,j)
        end do
    end do
    !$omp end parallel do

    !$omp parallel do schedule(static)
    do i = 1, LEN_2D
        a(i) = b(i) + c(i) * d(i)
    end do
    !$omp end parallel do

end subroutine tsvc_2_s2275_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s2275__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s2275_fp64(double * restrict a, double * restrict aa,
                       const double * restrict b, const double * restrict bb,
                       const double * restrict c, const double * restrict cc,
                       const double * restrict d, int64_t LEN_2D)
{
    #pragma omp parallel
    {
        #pragma omp for
        for (int64_t j = 0; j < LEN_2D; j++) {
            #pragma omp simd
            for (int64_t i = 0; i < LEN_2D; i++) {
                aa[j * LEN_2D + i] += bb[j * LEN_2D + i] * cc[j * LEN_2D + i];
            }
        }

        #pragma omp for
        for (int64_t i = 0; i < LEN_2D; i++) {
            a[i] = b[i] + c[i] * d[i];
        }
    }
}
